src/main/java/org/folio/innreach/domain/listener/KafkaInventoryEventListener.java

```diff
@@ -73,7 +80,13 @@ public void handleItemEvents(List<ConsumerRecord<String, DomainEvent<Item>>> con
   public void handleInstanceEvents(List<ConsumerRecord<String, DomainEvent<Instance>>> consumerRecords) {
     log.info("Handling inventory instance events from Kafka [number of events: {}]", consumerRecords.size());
     var events = getEvents(consumerRecords);
-
+    for(DomainEvent<Instance> event : events){
+      log.info("handleInstanceEvents:: Event type: {}, tenant: {}, timestamp: {}, data: {} ", event.getType(), event.getTenant(), event.getTimestamp(), event.getData());
+      var oldEntity = event.getData().getOldEntity();
+      var newEntity = event.getData().getNewEntity();
+      if(oldEntity!=null && newEntity!=null)
```

Comment:
And condition isnt the best thing to do. There can be a case that an oldEntity is null but newEntity is not null incase of new instance/hold/item created

Reply:
Added new condition.